### PR TITLE
Downloads

### PR DESCRIPTION
**Pull request by andrewkern**
this is a big update to the analysis 2 pipeline. I've done a few things on this PR

- added rules to automatically download and install tools we were asking the user to install manually
- added a `tiny_config.yml` as a very small workflow to use when developing or debugging
- cleaned up a bunch of bugs along the way. the workflow, as checking in on main, would not complete in my hands

in addition i'm really close to putting `smc++` back in the pipeline. i'm able to build it under `python 3.9` with our environment settings and i've put in the rules for the download / install. i'm just unsure if we want to use that as well? 

i'd like to get some other eyes on this PR if possible

### CODE REVIEW

**Comment by andrewkern** [comment]
so seems like UO server is faster. i'll write to Peter Keightley today and ask his permission to mirror that file.

**Comment by andrewkern** [comment]
any code review here @xin-huang or @chriscrsmith ?

**Comment by chriscrsmith** [comment]
sorry I don't get to it today, maybe next week

**Comment by xin-huang** [comment]
Reviewing, will continue tomorrow

**Review by xin-huang — COMMENTED** [review]
I can run `simualtion.snake` and `dfe.snake` successfully with `tidy_config.yaml`.

In `README.md`, I think we can remove `import numpy as np` and `.astype("int")`.
And `slim_scaling_factor=100` is too large, we could use `slim_scaling_factor=10`.

```
import stdpopsim

species = stdpopsim.get_species("HomSap")

# Adding a specific Distribution of fitness effects
dfe = species.get_dfe("Gamma_K17")

contig = species.get_contig("chr20")
model = species.get_demographic_model("OutOfAfrica_3G09")
samples = model.get_samples(100, 100, 100)  # YRI, CEU, CHB

# Extracting gene annottation
exons = species.get_annotations("ensembl_havana_104_exons")
exon_intervals = exons.get_chromosome_annotations("chr20")
contig.add_dfe(intervals=exon_intervals, DFE=dfe)

# Engine used for simulations
engine = stdpopsim.get_engine("slim")

# Simulate
ts = engine.simulate(
    model,
    contig,
    samples,
    seed=236,
    slim_scaling_factor=10,
    slim_burn_in=10,
)
```

**Comment by andrewkern** [comment]
I included comments from @xin-huang on the `README.md` here

**Review by chriscrsmith — APPROVED** [review]
Looks good! I'm only part way finished running the pipeline but it seems to be working well.

File: README.md
Comment (andrewkern) [reply README.md:99]: yeah i totally agree. i'm not going to delete it _yet_ but i think we might target deleting this section of the README in a future PR?

File: README.md
Comment (andrewkern) [reply README.md:83]: shouldn't matter if you use 10 or 30 cores here

Comment (chriscrsmith) [reply README.md:83]: sure I only commented because the preceding sentence said 10 cores i think

**Comment by chriscrsmith** [comment]
ran into a issue with braces in java commands inside workflows/stairway.py

L125: `f"{self.java_exe} -cp {self.classpath} {Stairway_plot_theta_estimation02} "`
L151: `f"{Stairway_plot_output_summary_commandline} {tmpdir} "`

I added those braces, and then snakemake just failed for me without explanation

**Comment by andrewkern** [comment]
interesting... will dig in here.

**Comment by andrewkern** [comment]
nice @chriscrsmith looks like there was a bug in here-- i had not updated the paths for `stairwayplot` to reflect my new change to move everything to `ext/` and it wasn't failing because i had an older version of the dir hanging out in `analysis2`.

i think there is only one path screwing this up, this one here: https://github.com/popsim-consortium/analysis2/blob/8d4acc589e801288da006b30fb982986e42d3320/workflows/n_t.snake#L186

**Comment by andrewkern** [comment]
okay got confirmation from @chriscrsmith that with this change we are GTG. squashed / rebased and will now wait for checks to pass before merge

**Pull request merged**
